Force more compatible transcoding profile for LiveTV

gnattu opened the pull request:
Live TV streams are unlikely to use modern features like HEVC and HDR, and the bitrate is also unlikely to be very high.

This sets a hard limit of 20 Mbps for Live TV transcoding and forces the use of the MPEG-TS container to solve compatibility issues with certain providers/tuners.

<!--
Ensure your title is short, descriptive, and in the imperative mood (Fix X, Change Y, instead of Fixed X, Changed Y).
For a good inspiration of what to write in commit messages and PRs please review https://chris.beams.io/posts/git-commit/ and our documentation.
-->

**Changes**
<!-- Describe your changes here in 1-5 sentences. -->

**Issues**
<!-- Tag any issues that this PR solves here.
ex. Fixes # -->

Fixes #11742
Fixes #11723

Shadowghost reviewed (approved):
Should be a good enough fix for most of the wonkyness

crobibero commented on Jellyfin.Api/Helpers/MediaInfoHelper.cs:
```diff
@@ -385,6 +385,21 @@ public void SortMediaSources(PlaybackInfoResponse result, long? maxBitrate)
+            // Remove all fmp4 transcoding profiles, because it causes playback error and/or A/V sync issues
+            // Notably: Some channels won't play on FireFox and LG webOs
+            // Some channels from HDHomerun will experience A/V sync issues
+            var profileForCompatibility = request.DeviceProfile.TranscodingProfiles.ToList();
+            profileForCompatibility.RemoveAll(p => p.Container == "mp4");
+            request.DeviceProfile.TranscodingProfiles = profileForCompatibility!.ToArray();
```
```suggestion
            request.DeviceProfile.TranscodingProfiles = request.DeviceProfile.TranscodingProfiles.Where(p => !string.Equals(p.Container, "mp4", StringComparison.OrdinalIgnoreCase)).ToArray();
```
Instead of creating list to just throw away

crobibero commented on Jellyfin.Api/Helpers/MediaInfoHelper.cs:
```diff
@@ -385,6 +385,21 @@ public void SortMediaSources(PlaybackInfoResponse result, long? maxBitrate)
+        // Enforce more restrictive transcoding profile for LiveTV due to compatability reasons
+        // Cap the MaxStreamingBitrate to 20Mbps, because we are unable to reliably probe source bitrate,
+        // which will cause the client to request extremely high bitrate that may fail the player/encoder
+        request.MaxStreamingBitrate = request.MaxStreamingBitrate > 20000000 ? 20000000 : request.MaxStreamingBitrate;
```
Note that if `MaxStreamingBitrate` is `null` it will remain `null`.

gnattu commented:
> introducing workarounds based on some arbitrary heuristics instead of actually addressing an issue is a secure path into code maintenance hell.

LiveTV is already a maintenance hell and all attempts to try to do things “correctly” will end up breaking working setups because everyone else is doing it wrong or at least, not ideal either.

RFC specs are being violated. Video frames are invalid. Timestamps go backwards during the stream. APIs lie about file type and format. These are just a few issues that come to mind, and there are many more we've encountered. Addressing these issues usually involves implementing various hacks because the system itself isn't functioning correctly. If another application works better, someone else may have hacked it more effectively than us, but still, it is still a hack. Our beloved VLC and MPV implemented lots of those hacks, but FFmpeg does not. 

For example, a common cause of the the A/V syncing issue from MPEG-TS streams is that some of the video or audio frames are missing or corrupted, and a video player optimized for corrupted streams will detect that frames are missing or invalid and the player will fill the missing frames (some will double the last or the next frame to adjust the sync), but ffmpeg’s mpeg-ts demuxer will not do this. So if you put that ts file contains bad frames into a player, it may look fine but if you convert it to mp4, it will eventually go out of sync.

Also, this does not work for recording because there is currently no easy way to tell if this file is a tv recording or not, which means there is no easy way to fix it in 10.9.

rimisuko commented:
Please excuse if the suggestion below is naive or stupid:

At least in my case here the audio sync problem arises when Jellyfin unnecessarily tries to remux a TS stream into fMP4. Provided it's possible to check whether a client is able to play a TS stream directly, checking the client's ability and skipping the remux if the client can handle a TS stream directly would address the issue safely and reliably for both live-tv and recordings.

Would ofc only work if it's possible to check the client's ability to play TS streams. But if so this would look like a relatively good solution to the issue to me. Maybe not as good as a correctly muxed fMP4 stream, but better than nothing.
